### PR TITLE
Explore: fix object value parsing for downloading traces as csv

### LINKED ISSUE #39039: CSV download of traces contains weird output
**What happened**:
I have downloaded CSV version of trace using Inspector in Exporer view.

Downloaded output seems to be not quite right with many `[object Object]` and ocasional `undefined` values.

```csv
"traceID","spanID","parentSpanID","operationName","serviceName","serviceTags","startTime","duration","logs","tags"
171c766b2e562646,171c766b2e562646,,HTTP POST - api_prom_push,cortex-gateway,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548250.929,175.399,undefined,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
171c766b2e562646,07f5e7c0c881f300,171c766b2e562646,Auth/Authenticate,cortex-gateway,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548251.0022,0.1,undefined,[object Object]
171c766b2e562646,7e4decea819718d2,171c766b2e562646,/httpgrpc.HTTP/Handle,cortex-gateway,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548255.5242,170.708,undefined,"[object Object],[object Object],[object Object]"
171c766b2e562646,63851360a3e4333d,7e4decea819718d2,/httpgrpc.HTTP/Handle,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548255.9841,169.907,"[object Object],[object Object],[object Object]","[object Object],[object Object],[object Object]"
171c766b2e562646,27d914c26baad16d,63851360a3e4333d,Distributor.biPushWrapper,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548256.607,169.269,undefined,[object Object]
171c766b2e562646,3977a284794b61b8,63851360a3e4333d,/cortex.Ingester/Push,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548260.0598,4.546,undefined,"[object Object],[object Object],[object Object]"
171c766b2e562646,6f36f47fc25c6af6,63851360a3e4333d,/cortex.Ingester/Push,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548260.253,4.273,undefined,"[object Object],[object Object],[object Object]"
171c766b2e562646,4dc3355544594841,63851360a3e4333d,/cortex.Ingester/Push,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548260.365,4.224,undefined,"[object Object],[object Object],[object Object]"
171c766b2e562646,7e61e7d3909ccd8d,63851360a3e4333d,/cortex.Ingester/Push,gme-distributor,"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]",1631187548260.492,4.138,undefined,"[object Object],[object Object],[object Object]"
```

**What you expected to happen**:
CSV output with real values.

**How to reproduce it (as minimally and precisely as possible)**:
Find a trace using Explorer view (I use Tempo datasource), click Inspector, Data, Download CSV.

**Environment**:
- Grafana version: Grafana v8.2.0-33442pre (e4ca6f2445)
- Data source type & version: Tempo
- OS Grafana is installed on: linux
- User OS & Browser: mac OS, Safari
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by tharun208**
Signed-off-by: tharun <rajendrantharun@live.com>

**What this PR does / why we need it**: Previously, values of type `object` is not formatted properly in the CSV export functionality for tempo. This PR fixes this issue by properly parsing the values of the `object` type before downloading it as CSV. 

**Which issue(s) this PR fixes**: #39039

Fixes #39039

**Special notes for your reviewer**:

### CODE REVIEW

File: packages/grafana-data/src/field/displayProcessor.ts [hunk 1]
```diff
@@ -168,7 +169,7 @@ function toStringProcessor(value: any): DisplayValue {
 
 export function getRawDisplayProcessor(): DisplayProcessor {
   return (value: any) => ({
-    text: `${value}`,
+    text: getFieldTypeFromValue(value) === 'other' ? `${JSON.stringify(value)}` : `${value}`,
```
Comment (mckn) [inline packages/grafana-data/src/field/displayProcessor.ts:172]: This processor will be used in more scenarios so I'm a bit hesitant regarding this change. Would love to get your input here @torkelo @ryantxu.

**Review by mckn — COMMENTED** [review]
Great work on this! I'm a bit concerned about the change in the displayProcessor but apart from that I think this is good.

File: packages/grafana-data/src/field/displayProcessor.ts (hunk 1 above)
Comment (ifrost) [reply packages/grafana-data/src/field/displayProcessor.ts:172]: Also - what if `JSON.stringify` throws an error (e.g. with an object with circular structure)?

Comment (tharun208) [reply packages/grafana-data/src/field/displayProcessor.ts:172]: @ifrost good catch, if we are fine with the above changes means, I can update it a replacerFunc to the `JSON.stringify` to fix the circular object issue.

**Comment by ifrost** [comment]
@tharun208 are you still working on this?

**Comment by tharun208** [comment]
> @tharun208 are you still working on this?

@ifrost I can update the JSON.stringify with a replacerFunc to fix the circular object issue. if you are good with this approach means I add this and commit

**Comment by ifrost** [comment]
> > @tharun208 are you still working on this?
> 
> @ifrost I can update the JSON.stringify with a replacerFunc to fix the circular object issue. if you are good with this approach means I add this and commit

I think it'd be good to add, but I would like someone from @grafana/user-essentials to approve these changes. I'm not sure what would be the impact of this change.

**Review by ifrost — APPROVED** [review]

**Review by ashharrison90 — APPROVED** [review]
looks good to me! 🙌

**Comment by ifrost** [comment]
@tharun208 Could you please sync up your branch with `main`? we had some issues with the build pipeline that were fixed by https://github.com/grafana/grafana/pull/45779

**Review by mckn — APPROVED** [review]
LGTM!

**Comment by ashharrison90** [comment]
great contribution @tharun208! 🙌

**Pull request merged**